src/mpid/ch4/src/ch4_part.h

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (C) by Argonne National Laboratory
+ *     See COPYRIGHT in top-level directory
+ */
+
+#ifndef CH4_PART_H_INCLUDED
+#define CH4_PART_H_INCLUDED
+
+#include "ch4_impl.h"
+
+MPL_STATIC_INLINE_PREFIX int MPIDI_part_start(MPIR_Request * request)
+{
+    int mpi_errno = MPI_SUCCESS;
+
+    MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDI_PART_STARTALL);
+    MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDI_PART_STARTALL);
+
+    MPID_THREAD_CS_ENTER(VCI, MPIDI_VCI(0).lock);
+    MPIDI_workq_vci_progress_unsafe();
```

Comment:
Same comment regarding vci lock and workq progress, and following functions.

Reply:
moved to MPIDIG